src/Take.Blip.Builder/Actions/ExecuteTemplate/ExecuteTemplateAction.cs

```diff
@@ -0,0 +1,111 @@
+        {
+            var result = string.Empty;
+            try 
+            {
+                var arguments = await GetScriptArgumentsAsync(context, settings, cancellationToken);
+                var obj = CopyArgumentsToObject(arguments);
+                var template = _handlebars.Compile(settings.Template);
+                result = template(obj);
+            }
+            catch (Exception ex)
+            {
+                if (ex is HandlebarsParserException)
+                {
+                    _logger.Error(ex, "Unexpected error while trying to parse Handlebars template");
+                    throw;
+                }
+
+                _logger.Error(ex, "Unexpected error while execute action Execute Template");
+                throw;
+            }
+            
+            await SetScriptResultAsync(context, settings, result, cancellationToken);
+        }
+
+        private async Task<JObject> GetScriptArgumentsAsync(
+            IContext context, ExecuteTemplateSettings settings, CancellationToken cancellationToken)
+        {
+            var obj = new JObject();
+            if (settings.InputVariables != null && settings.InputVariables.Length > 0)
+            {
+                for (int i = 0; i < settings.InputVariables.Length; i++)
+                {
+                    var variableValue = await context.GetVariableAsync(settings.InputVariables[i], cancellationToken);
+                    obj[settings.InputVariables[i]] = variableValue;
+                }
+            }
+            return obj;
+        }
+        
+        private async Task SetScriptResultAsync(
+            IContext context, ExecuteTemplateSettings settings, string result, CancellationToken cancellationToken)
+        {
+            if (result != null)
+            {
+                await context.SetVariableAsync(settings.OutputVariable, result, cancellationToken);
+            }
+            else
+            {
+                await context.DeleteVariableAsync(settings.OutputVariable, cancellationToken);
+            }
+        }
+        
+        private JObject CopyArgumentsToObject(JObject arguments)
+        {
+            var obj = new JObject();
+            foreach (var property in arguments.Properties())
+            {
+                var success = TryParseJson<JObject>(property.Value.ToString(), out var jObject);
+                if (success)
+                {
+                    foreach (var item in jObject)
+                    {
+                        obj[item.Key] = item.Value;
+                    }
+                }
+                else
+                {
+                    obj[property.Name] = property.Value;
+                }
+            }
+
+            return obj;
+        }
+        
+        private bool TryParseJson<T>(string json, out T result)
+        {
+            bool success = true;
+            var settings = new JsonSerializerSettings
```

Comment:
I built this method precisely to get the success value of the parse, since I don't want an exception to be thrown, because I already expect it to parse a string, so if I start the object in the constructor, I won't get this success value